R/xtab_corr.R

```diff
@@ -181,16 +181,20 @@ fei <- function(x, p = rep(1, length(x)),
                 ci = 0.95, alternative = "greater",
                 ...) {
   alternative <- .match.alt(alternative, FALSE)
-  table_dim <- dim(x)
-  is_1d_table <- is.null(table_dim) ||            # vector
-    length(table_dim) == 1 ||                     # 1D table
-    (length(table_dim) == 2 && table_dim[2] == 1) # matrix/data frame with 1 column
 
-  if (inherits(x, "BFBayesFactor") || !is_1d_table) {
+  if (inherits(x, "BFBayesFactor")) {
     insight::format_error("Fei is only applicable to goodness of fit tests.")
   } else if (!.is_htest_of_type(x, "Chi-squared test for given probabilities", "Chi-squared-test")) {
     x <- suppressWarnings(stats::chisq.test(x, y = NULL, p = p, rescale.p = TRUE))
     x$data.name <- NULL
+
+    table_dim <- dim(x$observed)
+    is_1d_table <- is.null(table_dim) ||            # vector
+      length(table_dim) == 1 ||                     # 1D table
+      (length(table_dim) == 2 && table_dim[2] == 1)
```

Comment:
I was just wondering, is it possible to have a 1D table with dim 1, 100 or similar? I.e. 1 in the first place? Then we should probably change `(length(table_dim) == 2 && table_dim[2] == 1)` to `(length(table_dim) == 2 && any(table_dim == 1))`